### ISSUE TITLE
Add HOWTO: Get per-example gradients

### ISSUE THREAD

**Issue by marcvanzee** [google/flax#858]
All layers accept batch dimension free inputs. Example:

```
def loss_fn(model, batch):
  return jnp.square(model * batch)

model = 3.
per_example_grad_fn = jax.vmap(jax.grad(loss_fn), in_axes=(None, 0))
per_example_grad_fn(model, jnp.ones((3,)))
```
